src/network/networkd-dhcp-common.c

```diff
@@ -1154,9 +1154,16 @@ int config_parse_duid_type(
 
         type = duid_type_from_string(type_string);
         if (type < 0) {
-                log_syntax(unit, LOG_WARNING, filename, line, type,
-                           "Failed to parse DUID type '%s', ignoring.", type_string);
-                return 0;
+                uint16_t t;
+
+                r = safe_atou16(type_string, &t);
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
+                                   "Failed to parse DUID type '%s', ignoring.", type_string);
+                        return 0;
+                }
+
+                type = t;
```

Comment:
Hmm, how do we know that `DUIDType` is wide enough? Maybe the type width needs to be forced:
```C
typedef enum DUIDType {
        DUID_TYPE_LLT       = 1,
        DUID_TYPE_EN        = 2,
        DUID_TYPE_LL        = 3,
        DUID_TYPE_UUID      = 4,
        _DUID_TYPE_MAX,
        _DUID_TYPE_INVALID  = -EINVAL,
        _DUID_TYPE_FORCE_U16 = UINT16_MAX,
} DUIDType;
```
See `_SD_ENUM_FORCE_S64`.

Please also add `assert(type == t)` here.